### PR TITLE
Refactor route viewer buttons to links

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
## Description

This PR refactors the route viewer buttons to display route shapes and pattern viewer into links.

I have added an underline effect when hovering the cursor over the links.
To simplify the code, "deselecting" a route once selected is removed.

## PR Checklist
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by daniel-heppner-ibigroup — APPROVED** [review]
Nice one!

**Comment by binh-dam-ibigroup** [comment]
> Code looks good, but behavior I have some issue with. The underline effect implies to the user that they're about to be navigated somewhere and while that may be strictly true it doesn't feel that way. Can we fade the background of the row or something? Maybe we can have the arrow appear? But the underline makes things confusing. Especially now that we can't click to de-select! Nothing happens when I click this link that appears as a link

@miles-grant-ibigroup How about I just remove the underline effect?

**Comment by miles-grant-ibigroup** [comment]
> > Code looks good, but behavior I have some issue with. The underline effect implies to the user that they're about to be navigated somewhere and while that may be strictly true it doesn't feel that way. Can we fade the background of the row or something? Maybe we can have the arrow appear? But the underline makes things confusing. Especially now that we can't click to de-select! Nothing happens when I click this link that appears as a link
> 
> @miles-grant-ibigroup How about I just remove the underline effect?

Could we replace it with some kind of background tint shift? I do like the feedback

**Comment by binh-dam-ibigroup** [comment]
> > > Code looks good, but behavior I have some issue with. The underline effect implies to the user that they're about to be navigated somewhere and while that may be strictly true it doesn't feel that way. Can we fade the background of the row or something? Maybe we can have the arrow appear? But the underline makes things confusing. Especially now that we can't click to de-select! Nothing happens when I click this link that appears as a link
> > 
> > 
> > @miles-grant-ibigroup How about I just remove the underline effect?
> 
> Could we replace it with some kind of background tint shift? I do like the feedback

@miles-grant-ibigroup Sure, I can do a background hover effect.

**Comment by binh-dam-ibigroup** [comment]
> > > Code looks good, but behavior I have some issue with. The underline effect implies to the user that they're about to be navigated somewhere and while that may be strictly true it doesn't feel that way. Can we fade the background of the row or something? Maybe we can have the arrow appear? But the underline makes things confusing. Especially now that we can't click to de-select! Nothing happens when I click this link that appears as a link
> > 
> > 
> > @miles-grant-ibigroup How about I just remove the underline effect?
> 
> Could we replace it with some kind of background tint shift? I do like the feedback

@miles-grant-ibigroup How about this: blue outline on hover, so it plays well with the keyboard focus outline? (see screenshot and https://github.com/opentripplanner/otp-react-redux/pull/1216/commits/a9005e311d2be396d4998529e6ec3b91315c142f). I thought about doing background+outline when active and background on hover in https://github.com/opentripplanner/otp-react-redux/pull/1216/commits/566e514d28d4bb5ed1ccafb4fdc53d62dd88b47e, but that messes up the keyboard focus.

image

**Comment by daniel-heppner-ibigroup** [comment]
Your screenshot looks really good @binh-dam-ibigroup. I didn't have a problem with the underline but this feels better!

File: lib/components/viewers/route-row.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ export const StyledRouteRow = styled.li`
 // Route Row Button sits invisible on top of the route name and info.
 export const RouteRowLink = styled(Link)`
   align-items: center;
+  border-radius: 4px;
```
Comment (miles-grant-ibigroup) [inline lib/components/viewers/route-row.tsx:40]: Any chance we can bump this up to maybe 10? Just to be in line with some of the rest of our UI

**Review by miles-grant-ibigroup — APPROVED** [review]
Looking good! One last style nitpick and then I think we're good to go

File: lib/components/viewers/route-row.tsx (hunk 1 above)
Comment (binh-dam-ibigroup) [reply lib/components/viewers/route-row.tsx]: Increassed to 10px in https://github.com/opentripplanner/otp-react-redux/pull/1216/commits/ec9b5527f61e24d4c9a952b5ab9d3754e0582d7d, so it is like the Nearby View.

**Pull request merged**